tests/Doctrine/Tests/DBAL/Functional/Schema/SqliteSchemaManagerTest.php

```diff
@@ -294,4 +294,26 @@ public function testOnlyOwnCommentIsParsed() : void
             ->getColumn('col1')
             ->getComment());
     }
+
+    /**
+     * @doesNotPerformAssertions
```

Comment:
Just a note: `@doesNotPerformAssertions` also disregards coverage (conscious decision on PHPUnit's side). The recommended way to test that an exception hasn't been thrown is calling `$this->addToAssertionCount(1)` after the method that throws exceptions.

Reply:
TIL

Reply:
@greg0ire https://github.com/sebastianbergmann/phpunit/pull/3348 :smile:

Reply:
I'm really confused about having `@doesNotPerformAssertions` here. Is the only outcome of actions in the test scenario is not having an exception thrown? I would understand such expectations in a validation scenario (no exception means “valid”) but not here.

Reply:
@morozov my goal here was to have a failing test for an existing bug, I'm not really trying to assert anything, I want to fix a crash, and make sure it does not come back. Tell me if this file is not the right place for that (since it is in `Functional`, it might not be, but I'm not sure where I should put this).

Reply:
> I want to fix a crash, and make sure it does not come back.

IMO, the purpose of a test is to ensure that the code works, not that it doesn't crash. I.e. should be worded in terms of assertions against produced results.

How do you know that just the absence of failure is enough in the reported case?

Reply:
Because this is what is complained about… they are not complaining that the fk does not get created properly for instance.

Anyway, I started by asserting that the generated SQL contains something about foreign key, but then I thought that even that might not work so I executed the SQL and asserted that there is a foreign key defined on it, tell me if I went overboard.

Reply:
> Because this is what is complained about… they are not complaining that the fk does not get created properly for instance.

I don't think this is the right mindset. But I believe we're on the same page now.

> […] tell me if I went overboard

The approach looks fine. The only question is, is this case is platform-specific or we can move it into a platform-independent suite and get covered on all platforms. The fact that people tend to write platform-specific tests because they experience issues using a given platform is the primary reason why we have inconsistency in platform support.